### PR TITLE
Fixes #28229 - upgrade babel and jest

### PR DESCRIPTION
**Pull request by sharvit**
**What this PR does:**
- Upgrade babel to v7
- Upgrade jest to v24
- Upgrade vendor to v3 (inlcudes a switch from `babel-polyfill` to `core-js`, see: https://babeljs.io/docs/en/babel-polyfill and https://github.com/theforeman/foreman-js/releases/tag/v3.0.0)
- Fixes tests caused by the upgrade

Blocked by https://github.com/theforeman/foreman/pull/6981

### CODE REVIEW

**Comment by jeremylenz** [comment]
Can this new babel config give us the ability to use
```
<>
...
</>
```

instead of 
```
<React.Fragment>
...
</React.Fragment>
```

?

(I've noticed it doesn't work currently and I think it was something to do with Babel understanding that JSX.)

**Comment by sharvit** [comment]
@jeremylenz I think it can be done with a babel plugin. I would prefer to do so in a different PR, maybe in the `env` package

File: package.json [hunk 1]
```diff
@@ -34,22 +34,22 @@
...
-    "@theforeman/vendor-dev": "^1.7.0",
+    "@theforeman/vendor-dev": "^3.0.0",
     "argv-parse": "^1.0.1",
     "axios-mock-adapter": "^1.10.0",
-    "babel-cli": "^6.10.1",
-    "babel-core": "^6.26.3",
-    "babel-eslint": "^8.2.3",
-    "babel-jest": "^23.6.0",
-    "babel-loader": "^7.1.1",
+    "@babel/cli": "^7.7.0",
```
Comment (evgeni) [inline package.json:40]: do we also need to package those? because we don't have babel 7.x packaged so far

Comment (evgeni) [reply package.json:40]: or is this just for dev setup?

Comment (sharvit) [reply package.json:40]: Didn't you packaged any `babel` before? You need them for the webpack build

Comment (sharvit) [reply package.json:40]: Notice there is a WIP to move them to a separate package:
https://github.com/theforeman/foreman/pull/7178

Comment (sharvit) [reply package.json:40]: The babel packages changed their names from v6 to v7
It used to be `babel-core` and now it's `@babel/core`

Comment (evgeni) [reply package.json:40]: and we only have babel-core (and friends) in 6.26 packaged, not 7.x

Comment (sharvit) [reply package.json:40]: True

**Comment by sharvit** [comment]
Instead of merging this PR, it makes more sense merging https://github.com/theforeman/foreman/pull/7178

**Comment by sharvit** [comment]
Closing, merged here: https://github.com/theforeman/foreman/pull/7178

**Pull request closed without merging**